src/data_management/rasters/RasterLayerTable.tsx

```diff
@@ -10,141 +10,182 @@ import {ExplainSideColumn} from '../../components/ExplainSideColumn';
-      } else{
-        return {...rowAllTables};
-      }
-    })
-    setTableData(tableDataDeletedmarker);
-    deleteRasters(uuids)
-    .then((_result) => {
-      // TODO: problem: triggerReloadWithCurrentPage requires a promise to set the checkboxes once the promise settles,
-      // but somehow triggerReloadWithCurrentPage is sometimes undefined leading to the error ".then of undefined"
-      // Workaround for now is to set the checkboxes before the promise returns.
-      // the function triggerReloadWithCurrentPage is actually the function fetchWithUrl 
-      // desired way would be:
-      // triggerReloadWithCurrentPage().then(()=>{
-      //   if (setCheckboxes) {
-      //     setCheckboxes([]);
-      //   }
-      // });
-      // workaround instead:
-      if (setCheckboxes) {
-        setCheckboxes([]);
-      }
-      triggerReloadWithCurrentPage();
-    })
+
+  const deleteActionRasters = (rows: any[], tableData:any, setTableData:any, triggerReloadWithCurrentPage:any, triggerReloadWithBasePage:any, setCheckboxes: any)=>{
+    setRowsToBeDeleted(rows);
+    const uuids = rows.map(row=> row.uuid);
+    setDeleteFunction(()=>()=>{
+      setBusyDeleting(true);
+      const tableDataDeletedmarker = tableData.map((rowAllTables:any)=>{
+        if (uuids.find((uuid)=> uuid === rowAllTables.uuid)) {
+          return {...rowAllTables, markAsDeleted: true}
+        } else{
+          return {...rowAllTables};
+        }
+      })
+      setTableData(tableDataDeletedmarker);
+      return deleteRasters(uuids)
+      .then((_result) => {
+        // TODO: problem: triggerReloadWithCurrentPage requires a promise to set the checkboxes once the promise settles,
+        // but somehow triggerReloadWithCurrentPage is sometimes undefined leading to the error ".then of undefined"
+        // Workaround for now is to set the checkboxes before the promise returns.
+        // the function triggerReloadWithCurrentPage is actually the function fetchWithUrl 
+        // desired way would be:
+        // triggerReloadWithCurrentPage().then(()=>{
+        //   if (setCheckboxes) {
+        //     setCheckboxes([]);
+        //   }
+        // });
+        setBusyDeleting(false);
+        // workaround instead:
+        if (setCheckboxes) {
+          setCheckboxes([]);
+        }
+        triggerReloadWithCurrentPage();
+        return new Promise((resolve, _reject) => {
+          resolve();
+        });
+      })
+    });
   }
-}
 
-const rasterSourceColumnDefenitions = [
-  {
-    titleRenderFunction: () => "Name",
-    renderFunction: (row: any) => 
-      <span
-        className={tableStyles.CellEllipsis}
-        title={row.name}
-      >
-        <NavLink to={`${navigationUrlRasters}/${row.uuid}/`}>{row.name}</NavLink>
-      </span>,
-    orderingField: "name",
-  },
-  {
-    titleRenderFunction: () =>  "Based on",
-    renderFunction: (row: any) => 
+  const rasterSourceColumnDefenitions = [
```

Comment:
one more comment, I noticed the name here is `rasterSourceColumnDefinitions` while it should be rasterLayer instead.

Reply:
done